### PR TITLE
Several HFS+ fixes

### PR DESCRIPTION
**Pull request by meder**
- Initialize HFS_INFO members earlier, since parsing code relies on them (#402)
- Do not bail when non-inline extended attributes are found
- Fix off-by-one bug

/cc @joachimmetz

### CODE REVIEW

File: tsk/fs/hfs.c [hunk 1]
```diff
@@ -2675,7 +2675,7 @@ hfs_attr_walk_special(const TSK_FS_ATTR * fs_attr,
     }
 
     // Allocate two buffers of the compression unit size.
-    rawBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE);
+    rawBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE + 1);
     uncBuf = (char *) tsk_malloc(COMPRESSION_UNIT_SIZE);
```
Comment (karlmortensen) [inline tsk/fs/hfs.c]: Why not add a +1 on the uncBuf size too?

Comment (bcarrier) [inline tsk/fs/hfs.c]: Does len end up being COMPRESSION_UNIT_SIZE+1 if the unit is not compressed and the new bounds checks prevent it from processing those?

The other minor change I'll make to this code after it is merged (or you can do it if you want) would be to allocate and free uncBuf and rawBuf closer to the loop that they are used it.  THey seem to be allocated early and freed late, which increases the number of places that we need to have error handling to free them.

**Comment by meder** [comment]
@karlmortensen i am not sure if it's needed there. the only time when i saw COMPRESSION_UNIT_SIZE + 1 being returned in len was when the COMPRESSION_UNIT_SIZE is uncompressed and 1 byte is added to indicate that. That is actually what the code further down on line 2866.

@bcarrier yes, len ended up being COMPRESSION_UNIT_SIZE + 1, but the bounds checking account for that. I'll push the uncBuf/rawBuf change shortly

**Comment by meder** [comment]
PTAL

**Pull request merged**